src/Storages/StorageReplicatedMergeTree.cpp

```diff
@@ -4177,6 +4177,11 @@ void StorageReplicatedMergeTree::startupImpl()
 
         /// In this thread replica will be activated.
         restarting_thread.start();
+        /// And this is just a callback
+        session_expired_callback_handler = EventNotifier::instance().subscribe(Coordination::Error::ZSESSIONEXPIRED, [this]()
```

Comment:
To be honest this doesn't matter, because it currently notifies manually from a Zookeeper client's internals only about this type of error. I added possibility to specify an event explicitly because this class could be valuable for other parts of code (not only for ReplicatedMergeTree)